packages/ozone-typescript-client/src/ozoneCredentials/ozoneCredentialsImpl.ts

```diff
@@ -13,7 +13,7 @@ export class SessionCredentials implements OzoneCredentials {
 				withCredentials: true
 			})
 		let authInfo = await (httpClient.call<AuthInfo>(request))
-		if (authInfo && !authInfo['principalId']) {
+		if (!authInfo || !authInfo['principalId']) {
```

Comment:
Do not see any reason why it was done like this
```suggestion
		if (!authInfo || !authInfo.principalId) {
```